test/test_client.py

```diff
@@ -1566,12 +1566,21 @@ def test_direct_connection(self):
         # direct_connection=False should result in RS topology.
         client = rs_or_single_client(directConnection=False)
         client.admin.command('ping')
-        self.assertGreaterEqual(len(client.nodes), 1)
+        nodes = client.nodes
+        self.assertGreaterEqual(len(nodes), 1)
         self.assertIn(client._topology_settings.get_topology_type(),
                       [TOPOLOGY_TYPE.ReplicaSetNoPrimary,
                        TOPOLOGY_TYPE.ReplicaSetWithPrimary])
         client.close()
 
+        # directConnection=True, should error with multiple hosts as a list.
+        if len(nodes) > 1:
+            hosts = []
+            for h, p in nodes:
+                hosts.append("%s:%s" % (h, p))
+            with self.assertRaises(ConfigurationError):
+                MongoClient(hosts, directConnection=True)
```

Comment:
I think we can use multiple hosts without caring whether they exist or not (since we'll raise an error before attempting to connect). So we could reduce this to:
```
            with self.assertRaises(ConfigurationError):
                MongoClient(['host1', 'host2'], directConnection=True)
```